boards/BMS/Src/App/App_Accumulator.c

```diff
@@ -194,3 +202,56 @@ void App_Accumulator_RunOnTick100Hz(struct Accumulator *const accumulator)
             break;
     }
 }
+
+bool App_Accumulator_CheckFaults(
+    struct BmsCanTxInterface *can_tx,
+    struct Accumulator *const accumulator,
+    bool                      isChargeState)
+{
+    bool overtemp_fault      = false;
+    bool undertemp_fault     = false;
+    bool overvoltage_fault   = false;
+    bool undervoltage_fault  = false;
+    bool communication_fault = App_Accumulator_HasCommunicationError(accumulator);
+
+    uint8_t throwaway_segment = 0U;
+    uint8_t throwaway_loc     = 0U;
+
+    float max_allowable_cell_temp = MAX_CELL_DISCHARGE_TEMP_DEGC;
+    float min_allowable_cell_temp = MIN_CELL_DISCHARGE_TEMP_DEGC;
+
+    // if we are charging, max cell temp is 45C not 60C
+    if (isChargeState)
```

Comment:
I think rather than checking if the charger is connected, we should check if TS current is negative (for charge current)